extension/src-language-server/snippets/stpa-snippets.ts

```diff
@@ -0,0 +1,382 @@
+        }
+    }
+    ControlledProcess {
+        hierarchyLevel 2
+        feedback {
+            [fb "feedback"] -> Sensor
+        }
+    }
+}
+`;
+        this.insertText = this.baseCode;
+    }
+}
+
+/**
+ * Snippet for a control structure with two consecutive controllers and one controlled process.
+ */
+export class ConsControllersSnippet extends STPALanguageSnippet {
+    constructor(documents: LangiumDocuments, shortId?: string) {
+        super(documents, "consControllerSnippet", shortId);
+        this.baseCode = `
+ControlStructure
+CS {
+    ControllerA {
+        hierarchyLevel 0
+        controlActions {
+            [ca "control action"] -> ControllerB
+        }
+    }
+    ControllerB {
+        hierarchyLevel 1
+        controlActions {
+            [ca "control action"] -> ControlledProcess
+        }
+        feedback {
+            [fb "feedback"] -> ControllerA
+        }
+    }
+    ControlledProcess {
+        hierarchyLevel 2
+        feedback {
+            [fb "feedback"] -> ControllerB
+        }
+    }
+}
+`;
+        this.insertText = this.baseCode;
+    }
+}
+
+/**
+ * Snippet for a control structure with two parallel controllers and one controlled process.
+ */
+export class ParallelContsSnippet extends STPALanguageSnippet {
+    constructor(documents: LangiumDocuments, shortId?: string) {
+        super(documents, "parallelContsSnippet", shortId);
+        this.baseCode =
+            `
+ControlStructure
+CS {
+    ControllerA {
+        hierarchyLevel 0
+        controlActions {
+            [ca "control action"] -> ControlledProcess
+        }
+    }
+    ControllerB {
+        hierarchyLevel 0
+        controlActions {
+            [ca "control action"] -> ControlledProcess
+        }
+    }
+    ControlledProcess {
+        hierarchyLevel 1
+        feedback {
+            [fbA "feedback"] -> ControllerA
+            [fbB "feedback"] -> ControllerB
+        }
+    }
+}
+`;
+        this.insertText = this.baseCode;
+    }
+}
+
+/**
+ * Snippet for a control structure with two consecutive controllers, one controlled process.
+ */
+export class ConsContsWithLoopSnippet extends STPALanguageSnippet {
+    constructor(documents: LangiumDocuments, shortId?: string) {
+        super(documents, "consContsWithLoopSnippet", shortId);
+        this.baseCode =
+            `
+ControlStructure
+CS {
+    ControllerA {
+        hierarchyLevel 0
+        controlActions {
+            [caC "control action"] -> ControlledProcess
+            [caB "control action"] -> ControllerB
+        }
+    }
+    ControllerB {
+        hierarchyLevel 1
+        controlActions {
+            [ca "control action"] -> ControlledProcess
+        }
+        feedback {
+            [fb "feedback"] -> ControllerA
+        }
+    }
+    ControlledProcess {
+        hierarchyLevel 2
+        feedback {
+            [fbA "feedback"] -> ControllerA
+            [fbB "feedback"] -> ControllerB
+        }
+    }
+}
+`;
```

Comment:
hardcoding the snippets into your ts files seems a little hacky, a collection of actual snippet files with one method to access them or add new ones seems a lot more generally accessible and expandible